### PR TITLE
Adding notes for using MQTT Server in a multi-tenant environment

### PR DESCRIPTION
**Pull request by nejraselimovic**
#### Summary
Adding notes about the difference in connecting to TTS MQTT Server for single- and multi-tenant environment.
Ref.: #280 

#### Notes for Reviewers
There is a lot of repeating info, but there is too much text and I can see that users can easily get lost. Any ideas for improvement?

#### Checklist
- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Run Locally: Verified that the docs build using `make server`, posted screenshots, verified external links. Test with `HUGO_PARAMS_SEARCH_ENABLED=true` if style changes will affect the search bar.
- [ ] New Features Marked: Documentation for new features is marked using the `new-in-version` shortcode, according to the guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [x] Style Guidelines: Documentation obeys style guidelines in [CONTRIBUTING](CONTRIBUTING.md).
- [ ] Commits: Commit messages follow guidelines in [CONTRIBUTING](CONTRIBUTING.md), there are no fixup commits left.

### CODE REVIEW

File: doc/content/integrations/mqtt-clients/eclipse-mosquitto/_index.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ For example, to subscribe to all topics in the application `app1`:
 $ mosquitto_sub -h thethings.example.com -t "#" -u app1 -P "NNSXS.VEEBURF3KR77ZR.." -d
 ```
 
+{{< note >}} For details about using this command in a multi-tenant environments like {{% tts %}} Cloud or Community Edition, see the [Multi-tenancy section of the MQTT Server guide](http://localhost:1313/integrations/mqtt#multi-tenancy). {{</ note >}}
```
Comment (johanstokking) [inline doc/content/integrations/mqtt-clients/eclipse-mosquitto/_index.md:36]: The Community Edition isn't multi-tenant really.

Basically, in every edition but The Things Stack Open Source, one needs to pass a tenant ID. But that doesn't mean the deployment is multi-tenant. TTN is a single tenant deployment.

Comment (johanstokking) [reply doc/content/integrations/mqtt-clients/eclipse-mosquitto/_index.md:36]: Also use `relref` or `ref`

File: doc/content/integrations/mqtt/_index.md [hunk 2]
```diff
@@ -13,7 +13,12 @@ weight: 30
 
 ## Multi-Tenancy
 
-Note that on multi-tenant environments such as Cloud, application IDs and other endpoints include the tenant ID, e.g `app1@tenant1`. On single tenant environments such as Open Source, the tenant ID can be removed, i.e `app1`.
+On multi-tenant environments such as Cloud, application IDs and other endpoints include the tenant ID, e.g `app1@tenant1`. On single-tenant environments such as Open Source, the tenant ID can be removed, i.e `app1`.
```
Comment (johanstokking) [inline doc/content/integrations/mqtt/_index.md:16]: TTN is single tenant and it cannot be removed.

I would say that one should always pass a tenant ID, except with The Things Stack Open Source.

**Comment by nejraselimovic** [comment]
I rearranged things in the MQTT guide in a way I think it's easier to follow, because I separated the generic info from examples (this was all mixed up). I also removed Cloud document completely, because all the info in there is already covered in MQTT guide and "special addressing" isn't just for the Cloud, but for all deployments except the Open Source (as @johanstokking mentioned above).

**Review by johanstokking — APPROVED** [review]
Thanks, I think this is very clear now.

Maybe hint that The Things Stack Community (Edition) / The Things Network uses tenant ID `ttn`

**Comment by Inrebusillis** [comment]
hi all
 I am stucked here and I thing I missed something obvious:
 I can subscribe MQTT for a TTN application using mosquito, MQTTlens, the thingsboard.
 However I cannnot publish any downlink.
  If I get an uplink with "V3/my_application@tnn/devices/dev_id/up" (and all is fine with all the systems)
  I assume that the topic for the downlink should be "V3/my_application@tnn/devices/dev_id/down/push"
  BUT
  I do not see even a message in the live data  of the gateway or application.
  When I use the downlink within the internal messaging function in TTN is working fine.
  
  ( tried all possible combination of tenant/multitenant etc....)
  not even get an error message...
I am  reading all the forum since 2 weeks, so please apologies me. I know is something simply that I do not see....
  thanks for your attention and support
  Best regards

**Comment by nejraselimovic** [comment]
> hi all I am stucked here and I thing I missed something obvious: I can subscribe MQTT for a TTN application using mosquito, MQTTlens, the thingsboard. However I cannnot publish any downlink. If I get an uplink with "V3/my_application@tnn/devices/dev_id/up" (and all is fine with all the systems) I assume that the topic for the downlink should be "V3/my_application@tnn/devices/dev_id/down/push" BUT I do not see even a message in the live data of the gateway or application. When I use the downlink within the internal messaging function in TTN is working fine.
> 
> ( tried all possible combination of tenant/multitenant etc....) not even get an error message... I am reading all the forum since 2 weeks, so please apologies me. I know is something simply that I do not see.... thanks for your attention and support Best regards

The correct topic to publish to for `app1` and `dev1` on `ttn` tenant would have the following format: `v3/app1@ttn/devices/dev1/down/push`. I see a typo in your message (`tnn` instead of `ttn` in your post) so that might be one of the many things that could go wrong. Btw you don't have to assume stuff like this, you just visit our [documentation site](https://www.thethingsindustries.com/docs/integrations/mqtt/), everything is explained there in detail.

**Comment by Inrebusillis** [comment]
thanks for your reply but is not that....( yeah typo is only the post)
I am sticking with the instruction of the documentation site but I get no message publish down...
Using  the internal native downlink messaging in TTN I got the node working and read the traffic in MQTTLens.
When I press the button "PUBLISH" in MQTT Lens nothing happen...
any chance that is someone reachable via telephone?
Thanks for your support and help
image

**Comment by Inrebusillis** [comment]
Thanks [nejraselimovic](https://github.com/nejraselimovic)!!
Your answer guided me to find the solution  digging deeper in the [documentation site](https://www.thethingsindustries.com/docs/integrations/mqtt/),
the issue was in the message and format that I sent, not in the topic!

Using this in JSON is working perfectly!
image
Thanks!!

**Comment by Inrebusillis** [comment]
Moving to next level...
I will be glad to know what I missing. 
I have two integration in the thingsboard one is linked to The Thing Network and one to The Thing Industries.
I have two  class C   electric plug capable to  switch the output OFF/ON
Same code everywhere.
The plug linked with TTN is working .
the plug linked with TTI is getting the upload data but not receiving the download. why???
 is there a different format in the downlink code to be used?
var data = {
            downlinks: [{
            f_port: 2,
            confirmed: false,
                   decoded_payload: {
                             "output": msg.params.replace(/"/g,"")
            },
            priority: "NORMAL"
        }]
    };

var result = {
    contentType: "JSON",
    data: JSON.stringify(data),
    metadata: { 
        devId : metadata.deviceName   //get devID from alias
    }
};
return result;


( may be is an issue for thingboard?)

**Comment by nejraselimovic** [comment]
> Moving to next level... I will be glad to know what I missing. I have two integration in the thingsboard one is linked to The Thing Network and one to The Thing Industries. I have two class C electric plug capable to switch the output OFF/ON Same code everywhere. The plug linked with TTN is working . the plug linked with TTI is getting the upload data but not receiving the download. why??? is there a different format in the downlink code to be used? var data = { downlinks: [{ f_port: 2, confirmed: false, decoded_payload: { "output": msg.params.replace(/"/g,"") }, priority: "NORMAL" }] };
> 
> var result = { contentType: "JSON", data: JSON.stringify(data), metadata: { devId : metadata.deviceName //get devID from alias } }; return result;
> 
> ( may be is an issue for thingboard?)

Please use [forum](https://www.thethingsnetwork.org/forum/) or [Slack](https://thethingsnetwork.slack.com/) channel for community support, GitHub issues shouldn't be the place for such discussions. However, if you think there is an issue with our documentation or there is a bug in The Things Stack implementation, you can always file an issue and we will investigate.

**Pull request merged**